### DISCUSSION TITLE
📔【操作系统】写时复制 Copy-on-write

### DISCUSSION THREAD

**Discussion by imageslr** [imageslr/blog#88] [Announcements]
https://imageslr.github.io/2020/copy-on-write.html 

 Copy-on-write 简介 写时复制（Copy-on-write，COW），有时也称为隐式共享（implicit sharing）COW 将复制操作推迟到第一次写入时进行：在创建一个新副本时，不会立即复制资源，而是共享原始副本的资源；当修改时再执行复制操作。通过这种方式共享资源，可以显著减少创建副本时的开销，同时，资源修改操作会增加少量开销。

**Comment by VoilaNeighbor** [comment]
C++98 标准并没有要求实现 CowString，也没有要求实现小对象优化。
